code/modules/surgery/organs/internal/eyes/_eyes.dm

```diff
@@ -629,44 +622,19 @@
 
 	if(QDELETED(eye_owner) || !ishuman(eye_owner)) //Other carbon mobs don't have eye color.
 		return
-
-	eye_owner.dna.species.handle_body(eye_owner)
-	update_mob_eyes_overlay()
-
-/**
- * Updates the emissive mob eye overlay
- *
- * When the light is on, the overlay(s) are added. When it is disabled, they are cut.
- * Adds one or two overlays depending on what the eye_color_mode toggle is set to.
- * Arguments:
- * * mob/living/carbon/eye_owner - the mob to add the overlay to
- */
-/obj/item/organ/internal/eyes/robotic/glow/proc/update_mob_eyes_overlay(mob/living/carbon/eye_owner = owner)
-	if(QDELETED(eye_owner))
-		return
-
-	if(!ishuman(eye_owner))
-		return
-
-	eye_owner.cut_overlay(eyes_overlay)
-	eye_owner.cut_overlay(eyes_overlay_left)
-	eye_owner.cut_overlay(eyes_overlay_right)
-
+	
 	if(!eye.on)
-		return
+		eye_icon_state = initial(eye_icon_state)
+		overlay_ignore_lighting = FALSE
+	else
+		overlay_ignore_lighting = TRUE
+		eye_icon_state = base_eye_state
 
-	switch(eye_color_mode)
-		if(MATCH_LIGHT_COLOR)
-			eyes_overlay = emissive_appearance('icons/mob/human/human_face.dmi', "eyes_glow_gs", eye_owner, layer = -BODY_LAYER, alpha = owner.alpha)
-			eyes_overlay.color = current_color_string
-			eye_owner.add_overlay(eyes_overlay)
-		if(USE_CUSTOM_COLOR)
-			eyes_overlay_left = emissive_appearance('icons/mob/human/human_face.dmi', "eyes_glow_gs_left", eye_owner, layer = -BODY_LAYER, alpha = owner.alpha)
-			eyes_overlay_right = emissive_appearance('icons/mob/human/human_face.dmi', "eyes_glow_gs_right", eye_owner, layer = -BODY_LAYER, alpha = owner.alpha)
-			eyes_overlay_left.color = eye_color_left
-			eyes_overlay_right.color = eye_color_right
-			eye_owner.add_overlay(eyes_overlay_left)
-			eye_owner.add_overlay(eyes_overlay_right)
+	var/obj/item/bodypart/head/head = eye_owner.get_bodypart(BODY_ZONE_HEAD) //if we have eyes we definently have a head anyway
+	var/previous_flags = head.head_flags
+	head.head_flags = previous_flags | HEAD_EYECOLOR
+	eye_owner.dna.species.handle_body(eye_owner)
+	head.head_flags = previous_flags
 
```

Comment:
this smells wrong what's going on here

Reply:
the HEAD_EYECOLOR flag controls whether eyes can be colored
moths do not have this flag so their eyes would not be colored if they changed the color of the eyes
this temporarily gives this flag so their eyes can be colored for the purpose of the luminescent eye light color
then calls a proc that updates their eyes among other overlay stuff
then sets its flags back to what it was before that

Reply:
what happens if something else calls handle_body in the background? what happens if flags are set inside handle_body and then overriden by previous_flags?
why can moths not have the flag? is it because of other colored eye overlays that don't fit them?
it seems like we need something more expansive to control what overlays are ok with what head then just this flag.

Reply:
> what happens if something else calls handle_body in the background?

shouldnt happen

> what happens if flags are set inside handle_body and then overriden by previous_flags?

handle_body doesnt change this flag

> why can moths not have the flag? is it because of other colored eye overlays that don't fit them?

their big bug eyes would look silly when colored

Reply:
also given that it touches the human variant too this could apply for any race that implants these